plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestRemoteQueryCommentLogging.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.postgresql;
+
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.trino.plugin.postgresql.PostgreSqlQueryRunner.createPostgreSqlQueryRunner;
+import static io.trino.tpch.TpchTable.CUSTOMER;
+import static io.trino.tpch.TpchTable.NATION;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestRemoteQueryCommentLogging
+        extends AbstractTestQueryFramework
+{
+    private TestingPostgreSqlServer postgreSqlServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        postgreSqlServer = closeAfterClass(new TestingPostgreSqlServer());
+        DistributedQueryRunner distributedQueryRunner = createPostgreSqlQueryRunner(
+                postgreSqlServer,
+                Map.of(),
+                Map.of("remote-query.comment.format", "query executed by $USER"),
+                List.of(CUSTOMER, NATION));
+
+        return distributedQueryRunner;
+    }
+
+    @Test(singleThreaded = true)
+    public void testShouldLogContextInComment()
+    {
+        assertThat(postgreSqlServer.recordEventsForOperations(() -> getQueryRunner().execute("CREATE TABLE postgresql.tpch.log_nation_test_table AS (SELECT * FROM postgresql.tpch.nation)"))
+                .stopEventsRecording()
```

Comment:
Instead of using the event number (counter) and using them to search for logs, can we use `executeWithQueryId`, configure queryId as a part of the comment and check if the event is logged or not  based on the queryId. It might take some time to iterate through all the logs and assert it - but I don't think it would be a bottleneck - ((Maybe we could configure ES test container for these logs )) ? Now we can split the test into smaller parts - for each type of query and they can run in parallel.

Reply:
Sure that might be a good idea, but let's do it as a followup

Reply:
Can we create an github issue for it ?

Reply:
Ok I will create one when this gets merged.

Reply:
We could even configure it to have automatic index in ES based on traceId - so have a quasi real life scenario in this tests - I really like this idea.

Reply:
Can we create a GitHub issue - for this one - So that we won’t forget this